### PR TITLE
[11.0][FIX] profiler module fixes

### PR DESCRIPTION
**Pull request by thomaspaulb**
- Exceptions raised within a profiled request were not dealt with and reraised properly at request end

- After exception was raised in the request we might not have a db cursor anymore, in that case save stats to file and log where the file is, so that you can still profile

- Postgresql profiling (for now) only used in 'full' mode

### CODE REVIEW

**Comment by thomaspaulb** [comment]
@sbidoul I changed `dns` to `dnspython`, and it seems to install fine, yet is [not found by Odoo](https://runboat.odoo-community.org/api/v1/builds/b60fd2e87-f1c4-40e8-a182-788a8dc9b6ae/init-log). What am I doing wrong?

**Comment by sbidoul** [comment]
`dns` was fine. What's needed is an external_dependencies_override in setup.py. See setuptools-odoo docs for an example. Fortunately this kind of thing has become unnecessary in Odoo 12+ (or 13 not sure).

**Comment by thomaspaulb** [comment]
Are you sure that `dns` is fine?

```
(.venv) $ python --version
Python 3.5.10
(.venv) $ pip install dns
ERROR: Could not find a version that satisfies the requirement dns
ERROR: No matching distribution found for dns
(.venv) $ pip install dnspython
Collecting dnspython
  Downloading dnspython-1.16.0-py2.py3-none-any.whl (188 kB)
     |████████████████████████████████| 188 kB 5.4 MB/s 
Installing collected packages: dnspython
Successfully installed dnspython-1.16.0
```

**Comment by sbidoul** [comment]
Sorry, I mean dns is fine in he manifest external_dependencies. But it must be mapped to dnspython in setup.py.

**Comment by thomaspaulb** [comment]
Thanks!

File: setup/letsencrypt/setup.py [hunk 1]
```diff
@@ -2,5 +2,14 @@
...
     setup_requires=['setuptools-odoo'],
-    odoo_addon=True,
+    odoo_addon={
+        'external_dependencies_override': {
+            'python': {
+                'acme': 'acme',
+                'cryptography': 'cryptography',
+                'dns': 'dnspython==1.16.0',
+                'josepy': 'josepy',
+            }
```
Comment (sbidoul) [inline setup/letsencrypt/setup.py:12]: Only dns is necessary here. Also, don't pin version unless necessary. Or use a lower bound.

**Review by sbidoul — COMMENTED** [review]
I realize I had written but not submitted this comment.

**Pull request closed without merging**